python/dgl/distributed/partition.py

```diff
@@ -23,16 +24,25 @@
     ETYPE: F.int32
     }
 
-def _save_graphs(filename, g_list):
-    '''Format data types in graphs before saving
+def _save_graphs(filename, g_list, formats=None, sort_etypes=False):
+    '''Preprocess partitions before saving:
+    1. format data types.
+    2. sort csc/csr by tag.
     '''
     for g in g_list:
         for k, dtype in RESERVED_FIELD_DTYPE.items():
             if k in g.ndata:
                 g.ndata[k] = F.astype(g.ndata[k], dtype)
             if k in g.edata:
                 g.edata[k] = F.astype(g.edata[k], dtype)
-    save_graphs(filename , g_list)
+    for g in g_list:
+        if (not sort_etypes) or (formats is None):
```

Comment:
Would `formats=None` but graph has csr or csc so that the formats are saved without sorting?

Reply:
if `formats=None`, only `COO` is saved. The case you mentioned should never happen I think.

Reply:
Is it possible that the pass-in graph only has csr/coo formats (e.g., using `g.format` API). When the `formats` argument is None, the default behavior is to select any existing format which can be csr/coo in this case.

Reply:
If the input graph has only `CSC/CSR`, partition cannot be done. `COO` is required. This is because `g.find_edges(eids)` is called, I think. And a graph with `coo` not allowed is quite rare, I think.

Reply:
Then could you add a check and a proper error message to forbid that?

Reply:
added.